File: examples/src/main/scala/sample/scala/parameterizedtype/generic/Stacks.scala
package sample.scala.parameterizedtype.generic

object Stacks {
  def reverse[T](in:Stack[T]):Stack[T] = {
    val out = new ListStack[T]
    while(!in.empty){
      val elt = in.pop
      out.push(elt)
    }
    return out
  }
}

File: examples/src/main/scala/sample/java/parameterizedtype/generic/Stacks.java
package sample.java.parameterizedtype.generic;

class Stacks {
    public static <T> Stack<T> reverse(Stack<T> in) {
      Stack<T> out = new ArrayStack<T>();
      while (!in.empty()) {
        T elt = in.pop();
        out.push(elt);
      }
      return out;
    }
}